test, deps: migrate from jest to vitest

0b5vr opened the pull request:
much faster tests!

ref, custom matcher: https://vitest.dev/blog/vitest-3-2.html#the-new-multi-purpose-matchers-type

0b5vr committed:
test, deps: migrate from jest to vitest

much faster tests!

ref, custom matcher: https://vitest.dev/blog/vitest-3-2.html#the-new-multi-purpose-matchers-type

yue4u commented on vitest.config.mjs:
```diff
@@ -0,0 +1,11 @@
+import { defineConfig } from 'vitest/config';
```
Maybe we can use vitest workspace(projects) support to avoid duplicate vitest config?

https://vitest.dev/guide/projects

0b5vr replied:
tried @ https://github.com/pixiv/three-vrm/pull/1770/commits/a62c7a9e6d4042bc57634c963c84a5c7b12d2393

0b5vr committed:
test: move vitest config to project config

Removed test script from individual packages.
To run tests for individual packages, run like this: `yarn test --project @pixiv/three-vrm-core`

Ref: https://vitest.dev/guide/projects

See: https://github.com/pixiv/three-vrm/pull/1770#discussion_r2646629279

0b5vr committed:
test, deps: add @vitest/coverage-v8

also add coverage artifact directory to .gitignore

coverage is not included in the CI though

ke456-png reviewed (approved):
Looks good.
